### PR TITLE
Drop support for Node.js < 4

### PR DESCRIPTION
**Pull request by realityking**
Also updates dev dependencies to their latest version.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,7 +1,9 @@
 language: node_js
 node_js:
-  - 0.8
-  - 0.10
+  - 4
+  - 6
+  - 8
+  - 'stable'
```
Comment (honzajavorek) [inline .travis.yml]: Ha! This is good. We could be using it also with the Dredd repo.

File: package.json [hunk 2]
```diff
@@ -16,9 +19,9 @@
     "http-string-parser": "0.0.6"
   },
   "devDependencies": {
-    "coffee-script": "1.6.3",
-    "mocha": "1.11.0",
-    "chai": "1.7.1"
+    "chai": "^4.1.2",
+    "coffeescript": "^1.12.7",
+    "mocha": "^4.0.1"
```
Comment (honzajavorek) [inline package.json]: There's a new process with us which requires us to use only approved dependencies. That basically means the deps' licensing needs to be all right. For this reason we're not relaxing the versions anymore.

To be honest, I wasn't much fan of the relaxed versions anyway. To me as a maintainer, they bring more hassle than benefit and it felt to me it's always better to just explicitly upgrade the deps, e.g. with the help of https://greenkeeper.io/

Would you please keep the versions pinned?

**Comment by realityking** [comment]
Updated with pinned versions. Sorry it took so long.

Setting up GreenKeeper sounds like a good idea with this new constraint.

File: package.json [hunk 3]
```diff
@@ -16,9 +19,9 @@
     "http-string-parser": "0.0.6"
   },
   "devDependencies": {
-    "coffee-script": "1.6.3",
-    "mocha": "1.11.0",
-    "chai": "1.7.1"
+    "chai": "4.1.2",
+    "coffeescript": "1.12.7",
+    "mocha": "5.0.0"
```
Comment (honzajavorek) [inline package.json]: I'm sorry this takes so long. We need to get this https://github.com/kumavis/browser-stdout/pull/4 resolved first. Current Mocha has a license issue - their dependency tree contains the `browser-stdout` library, which has invalid license. I'll get back to this as soon as possible.

Comment (honzajavorek) [reply package.json]: Filed https://github.com/mochajs/mocha/issues/3248

Comment (realityking) [reply package.json]: Ah, that's a bummer. Looks like that dependency was added [this commit](https://github.com/mochajs/mocha/commit/15c344c067884cf4856bf0c4235d81be688d54d5) which was first part of [version 3.0](https://github.com/mochajs/mocha/releases/tag/v3.0.0).

Would it help if I used mocha 2.5.3 instead? That's the last release before this dependency was added.

Comment (honzajavorek) [reply package.json]: There is some activity on the Mocha issue, so let's see how it turns out. I'd better upgrade than downgrade, if possible. The same problem hurts Dredd, etc.

Comment (realityking) [reply package.json]: `browser-stdout` 1.3.1 was released - now with a license file 🎉

Comment (honzajavorek) [reply package.json]: https://github.com/mochajs/mocha/pull/3258

**Pull request merged**